lld/ELF/Arch/RISCV.cpp

```diff
@@ -650,8 +650,36 @@ static void relaxHi20Lo12(const InputSection &sec, size_t i, uint64_t loc,
   if (!gp)
     return;
 
+  bool noRelaxation = false;
```

Comment:
Do we need this variable? Can we use `!isInt<12>(r.sym->getVA(r.addend) - gp->getVA())` in the one place we use the variabe?

Reply:
Sure. This was added in anticipation of another reason we might decide not to relax that is added in  [this PR](https://github.com/llvm/llvm-project/pull/73624)